tycho-api/src/main/java/org/eclipse/tycho/ReproducibleUtils.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tycho;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for reproducible builds.
+ */
+public class ReproducibleUtils {
+    private ReproducibleUtils() {
+    }
+
+    /**
+     * Writes the property list to the output stream in a reproducible way. The java.util.Properties
+     * class writes the lines in a non-reproducible order, adds a non-reproducible timestamp and
+     * uses platform-dependent new line characters.
+     * 
+     * @param properties
+     *            the properties object to write to the file.
+     * @param file
+     *            the file to write to. All the missing parent directories are also created.
+     * @throws IOException
+     *             if writing the property list to the specified output stream throws an
+     *             IOException.
+     */
+    public static void storeProperties(Properties properties, File file) throws IOException {
```

Comment:
```suggestion
    public static void storeProperties(Properties properties, Path file) throws IOException {
```

Just a nitpick here, but for new API better always use `Path` as `File` can trivially be converted into a `Path`.

Reply:
We have a File object in all the call sites, so using a Path will just add code for nothing. Also I prefer to use File because the name is more clear and File objects are simpler to create. Path does not supersede File, I don't think it's necessarily better to use it on new code.

Reply:
Maven4 is migrating all `File` > `Path` as well and as you already converted the file into path in this method it seems suitable here as well. I also don't see how `new File("...")` is simpler than `Path.of("...")`... and path is usually more flexible as it is not limited to local files.